selinux/cephmetrics.te

```diff
@@ -1,32 +1,45 @@
-policy_module(cephmetrics, 1.0.0)
+policy_module(cephmetrics 1.1.0)
+
+# Build requires selinux devel package to be installed.
+# 1. copy this 'te' file to /tmp
+# 2. cd /tmp
+# 3. make -f /usr/share/selinux/devel/Makefile cephmetrics.pp
+# 4. semodule -i /tmp/cephmetric.pp
 
 require {
-	type collectd_t;
-	type ceph_t;
-	type ceph_var_run_t;
-	type ceph_var_lib_t;
-	type fixed_disk_device_t;
-	class unix_stream_socket connectto;
-	class dir read;
-	class file getattr;
-	class capability2 block_suspend;
+
+        type collectd_t;
+        type ceph_t;
+        type ceph_exec_t;
+        type bin_t;
+        type tmp_t;
+        type ceph_var_run_t;
+        type ceph_var_lib_t;
+        type var_log_t;
+        type fixed_disk_device_t;
+
+        class unix_stream_socket connectto;
+        class dir { read write };
+        class capability2 block_suspend;
+        class file { getattr execute execute_no_trans open read write };
 }
 
 #============= collectd_t ==============
 
 #!!!! This avc can be allowed using the boolean 'daemons_enable_cluster_mode'
+allow collectd_t bin_t:file { execute execute_no_trans };
 allow collectd_t ceph_t:unix_stream_socket connectto;
+allow collectd_t ceph_exec_t:file { execute execute_no_trans open read };
 allow collectd_t ceph_var_run_t:dir read;
-allow collectd_t ceph_var_lib_t:file getattr;
-allow collectd_t ceph_var_lib_t:lnk_file getattr;
-allow collectd_t ceph_var_lib_t:lnk_file read;
+allow collectd_t ceph_var_lib_t:file { getattr open read };
+allow collectd_t ceph_var_lib_t:lnk_file {getattr open read };
 allow collectd_t fixed_disk_device_t:blk_file getattr;
 allow collectd_t self:capability2 block_suspend;
 allow collectd_t var_log_t:dir { add_name write };
-allow collectd_t var_log_t:file create;
+allow collectd_t var_log_t:file { create open write };
+
 corecmd_exec_shell(collectd_t)
 files_list_tmp(collectd_t)
 libs_exec_ldconfig(collectd_t)
 libs_manage_lib_dirs(collectd_t)
 libs_manage_lib_files(collectd_t)
-logging_write_generic_logs(collectd_t)
```

Comment:
Are you sure we no longer need this bit?

Reply:
I've updated to use the standard policies (logging_write_generic_logs)  instead of listing rules explicitly (through allows)
The updated pp is running on Lubos's QE environment now